infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/logreader/BaseSnapshotReader.java

```diff
@@ -141,24 +141,23 @@ private SMREntryList next(OpaqueStreamIterator stream) {
         int currentMsgSize = 0;
 
         try {
-            while (currentMsgSize < maxDataSizePerMsg) {
+            while (currentMsgSize < maxTransferSize) {
                 if (lastEntry != null) {
                     List<SMREntry> smrEntries = lastEntry.getEntries().get(stream.uuid);
                     if (smrEntries != null) {
                         int currentEntrySize = ReaderUtility.calculateSize(smrEntries);
-
                         if (currentEntrySize > DEFAULT_MAX_DATA_MSG_SIZE) {
```

Comment:
We should enhance this check now?
This check now makes sense only if maxTransferSize > DEFAULT_MAX_DATA_MSG_SIZE. Otherwise LR will only support moving records which are at the max `maxTransferSize`

Reply:
The recommended size limit for a protobuf message is 64MB.  Here we are checking against that and failing the read if the size is beyond 64MB.  On L153, the size is compared against the transfer size.  I think both checks are valid?  Let me know if I did not follow your comment fully...

Reply:
if the data size is less than 64MB but say more than maxTransferSize, we will be stuck in this loop forever. I am thinking about throwing an exception and maybe putting that particular session in the STOPPED state. That way other sessions will not be affected (more so with the thread pool change. Less starvation)

Reply:
As of now, the plan is to never expose the runtime uncompressed tx size so it will be 100MB and always more than the max data msg size(64 MB).  

But in future if it changes, this is a valid point.  So added the fix.

Also, the ideal fix here should be to split the transaction as currentEntrySize is calculated for all smr entries in the transaction.  As it will be a basic functional change, I am not adding it now and just put a TODO for it.